watcher/local/watcher.go

```diff
@@ -15,12 +15,92 @@
 package local
 
 import (
+	"context"
+	"sync"
+
+	"github.com/pkg/errors"
+
 	"perun.network/go-perun/channel"
+	"perun.network/go-perun/watcher"
 )
 
 type (
+	// Watcher implements a local watcher.
+	Watcher struct {
+		rs channel.RegisterSubscriber
+		*registry
+	}
+
+	// Config represents the configuration for initializing a local watcher.
+	Config struct {
+		RegisterSubscriber channel.RegisterSubscriber
+	}
+
 	ch struct {
 		id     channel.ID
+		params *channel.Params
 		parent channel.ID
+
+		subChsAccess sync.Mutex
```

Comment:
Can you please document when this mutex needs to he locked? So to say, what does it guard?

Reply:
moved to #213